### PR TITLE
Create example application for react native sdk

### PR DESCRIPTION
**Pull request by vishnumad**
### _Summary_
Creates an example native Dapp for React Native

### _How did you test your changes?_

image

### CODE REVIEW

File: react-native/example/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "private": true,
+  "scripts": {
+    "android": "react-native run-android",
+    "ios": "react-native run-ios",
+    "start": "react-native start",
+    "test": "jest",
+    "lint": "eslint ."
+  },
+  "dependencies": {
+    "@coinbase/wallet-mobile-sdk": "0.3.1",
```
Comment (bangtoven) [inline react-native/example/package.json]: i guess we'd want to point this to local path instead. how do you think?

Comment (vishnumad) [reply react-native/example/package.json]: Changed to point to local path. [2d6b93d](https://github.com/coinbase/wallet-mobile-sdk/pull/47/commits/2d6b93d96e70cf2eeb10f5ad896c75a9697f869e)
One thing to note here is that when testing, every time a change is made in the RN sdk, you'll need to re-run `yarn add 'file:..'`

**Review by bangtoven — DISMISSED** [review]

File: react-native/example/package.json (hunk 1 above)
Comment (bangtoven) [reply react-native/example/package.json]: 👍

File: react-native/example/App.js [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+  const backgroundStyle = {
+    backgroundColor: Colors.lighter,
+  };
+
+  return (
+    <SafeAreaView style={backgroundStyle}>
+      <StatusBar />
+      <ScrollView style={styles.scrollViewStyle}>
+        <Section title="Methods">
+          {!isConnected && (
```
Comment (AMITGOELNY) [inline react-native/example/App.js]: nit: is it possible to do ternary assignment on this?

File: react-native/example/android/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+    }
+}
+
+dependencies {
+    implementation fileTree(dir: "libs", include: ["*.jar"])
+
+    //noinspection GradleDynamicVersion
+    implementation "com.facebook.react:react-native:+"  // From node_modules
+
+    implementation "androidx.swiperefreshlayout:swiperefreshlayout:1.0.0"
```
Comment (AMITGOELNY) [inline react-native/example/android/app/build.gradle:257]: is this one required?

File: react-native/example/android/app/src/debug/java/com/example/ReactNativeFlipper.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            public void apply(OkHttpClient.Builder builder) {
+              builder.addNetworkInterceptor(new FlipperOkhttpInterceptor(networkFlipperPlugin));
+            }
+          });
+      client.addPlugin(networkFlipperPlugin);
+      client.start();
+
+      // Fresco Plugin needs to ensure that ImagePipelineFactory is initialized
+      // Hence we run if after all native modules have been initialized
+      ReactContext reactContext = reactInstanceManager.getCurrentReactContext();
```
Comment (AMITGOELNY) [inline react-native/example/android/app/src/debug/java/com/example/ReactNativeFlipper.java:52]: do we need all these plugins for flipper, or flipper at all?

File: react-native/example/android/app/src/main/java/com/example/MainApplication.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+import com.facebook.react.ReactInstanceManager;
+import com.facebook.react.ReactNativeHost;
+import com.facebook.react.ReactPackage;
+import com.facebook.react.config.ReactFeatureFlags;
+import com.facebook.soloader.SoLoader;
+import com.example.newarchitecture.MainApplicationReactNativeHost;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
+public class MainApplication extends Application implements ReactApplication {
```
Comment (AMITGOELNY) [inline react-native/example/android/app/src/main/java/com/example/MainApplication.java:19]: are we locked into java for this?  Is there an option to use kotlin?

File: react-native/example/android/app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <string name="app_name">example</string>
```
Comment (AMITGOELNY) [inline react-native/example/android/app/src/main/res/values/strings.xml]: may want to have app name thats displayed in app drawer say something else, as example is pretty vague for an app name.  Perhaps, Coinbase Wallet SDK Example

File: react-native/example/android/app/build.gradle (hunk 3 above)
Comment (vishnumad) [reply react-native/example/android/app/build.gradle:257]: All the native code here is auto-generated from the `npx create-react-native-app` script

File: react-native/example/android/app/src/debug/java/com/example/ReactNativeFlipper.java (hunk 4 above)
Comment (vishnumad) [reply react-native/example/android/app/src/debug/java/com/example/ReactNativeFlipper.java:52]: Auto generated code

File: react-native/example/android/app/src/main/java/com/example/MainApplication.java (hunk 5 above)
Comment (vishnumad) [reply react-native/example/android/app/src/main/java/com/example/MainApplication.java:19]: Auto generated code

File: react-native/example/App.js (hunk 2 above)
Comment (vishnumad) [reply react-native/example/App.js]: I find this a little easier to parse visually, but happy to change to a ternary if you feel strongly about it

Comment (bangtoven) [reply react-native/example/App.js]: i also think that the current code is more in line with js convention.

**Review by bangtoven — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for bangtoven @ 2022-09-08 23:18:34 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Review by bangtoven — APPROVED** [review]

File: react-native/example/android/app/src/debug/java/com/example/ReactNativeFlipper.java (hunk 4 above)
Comment (AMITGOELNY) [reply react-native/example/android/app/src/debug/java/com/example/ReactNativeFlipper.java:52]: same as above.  understood this is generated, but if unneeded, we may want to remove to reduce noise.

File: react-native/example/android/app/build.gradle (hunk 3 above)
Comment (AMITGOELNY) [reply react-native/example/android/app/build.gradle:257]: gotcha.  not sure if the scaffolding is adding extra noise for what's needed, or adding extra unused dependencies.  In either case, we can clean up later..

**Review by AMITGOELNY — APPROVED** [review]

**Pull request merged**